tests/Feature/Controllers/Api/ProjectControllerTest.php

```diff
@@ -0,0 +1,566 @@
+            'title' => 'ab',
+            'description' => 'ab',
+        ]);
+
+        $this->minimumLengthValidation('post', '/api/v1/projects', $data);
+
+        $this->assertEquals(0, Project::count());
+    }
+
+    public function test_the_provided_title_and_description_cannot_be_longer_than_255_characters()
+    {
+        $data = array_merge($this->data, [
+            'title' => str_repeat('abc', 90),
+            'description' => str_repeat('abc', 90),
+        ]);
+
+        $this->maximumLengthValidation('post', 'api/v1/projects', $data);
+
+        $this->assertEquals(0, Project::count());
+    }
+
+    public function test_the_provided_due_date_must_be_greater_than_the_current_date_when_creating_a_project()
+    {
+        $data = array_merge($this->data, [
+            'due_date' => now()->format('Y-m-d'),
+        ]);
+
+        $this->dateAfterValidation('POST', 'api/v1/projects', $data);
+
+        $this->assertEquals(0, Project::count());
+    }
+
+    public function test_the_provided_due_date_must_be_before_january_first_2031_when_creating_a_project()
+    {
+        $data = array_merge($this->data, [
+            'due_date' => Carbon::create(2031, 1, 1),
+        ]);
+
+        $this->dateBeforeValidation('POST', 'api/v1/projects', $data);
+
+        $this->assertEquals(0, Project::count());
+
+    }
+
+    public function test_a_guest_cannot_access_the_create_project_endpoint()
+    {
+        $this->postJson('/api/v1/projects', $this->data)
+            ->assertUnauthorized();
+
+        $this->assertEquals(0, Project::count());
+    }
+
+    public function test_a_user_without_the_create_project_permission_cannot_access_the_create_project_endpoint()
+    {
+        $this->actingAs($this->employee)->postJson('api/v1/projects', $this->data)
+            ->assertForbidden();
+
+        $this->assertEquals(0, Project::count());
+    }
+
+    public function test_a_user_with_the_create_project_permission_can_access_the_create_project_end_point()
+    {
+        $this->actingAs($this->manager)->postJson('api/v1/projects', $this->data)
+            ->assertCreated()
+            ->assertJsonFragment([
+                'title' => $this->data['title'],
+                'description' => $this->data['description'],
+                'due_date' => Carbon::parse($this->data['due_date'])->format('d M Y'),
+            ]);
+
+        $this->assertEquals(1, Project::count());
+    }
+
+    public function test_a_guest_cannot_access_the_project_detail_endpoint()
+    {
+        $project = Project::factory()->create();
+
+        $this->getJson('/api/v1/projects/'.$project->id)
+            ->assertUnauthorized();
+    }
+
+    public function test_a_user_without_the_read_project_permission_cannot_access_the_project_detail_endpoint()
+    {
+        $project = Project::factory()->create();
+
+        $this->actingAs($this->user)->getJson('/api/v1/projects/'.$project->id)
+            ->assertForbidden();
+    }
+
```

Comment:
test a user with the read project permission can access the project detail endpoint is vergeten